sunpy/map/header_helper.py

```diff
@@ -384,8 +384,8 @@ def _set_instrument_meta(meta_wcs, instrument, telescope, observatory, detector,
     'Matrix element CDi_j describing the rotation required to align solar North with the top of the image.'
 }
 
-
-def make_heliographic_header(date, observer_coordinate, shape, *, frame, projection_code="CAR"):
+@u.quantity_input
+def make_heliographic_header(date, observer_coordinate, shape, *, frame, projection_code="CAR", map_center_longitude=0.0*u.deg):
```

Comment:
```suggestion
def make_heliographic_header(date, observer_coordinate, shape, *, frame, projection_code="CAR",
                             map_center_longitude: u.Quantity['angle'] = 0.0*u.deg):
```
You specified the argument to have a `Quantity` as the default value, but that doesn't actually specify that the argument has to be a `Quantity` or what units are allowed.  You need to specify what unit(s) are permitted so that the input can be validated.

The `UnitsError`s that you were getting were from subsequent code trying to use the provided value, rather than from validation of the inputs to this function.